### PR TITLE
MULE-16581: Runtime does not shutdown after OptionalLong[15000] milliseconds

### PR DESCRIPTION
**Pull request by plbalbi**

### CODE REVIEW

File: modules/service/src/main/java/org/mule/runtime/module/service/internal/manager/MuleServiceManager.java [hunk 1]
```diff
@@ -67,15 +70,35 @@ private void startServices() throws MuleException {
...
 
+  /**
+   * Stop all discovered services. In the case of 'HTTP' and 'Scheduler' service, if present, they should be stopped last and in
+   * that order.
+   *
+   * @throws MuleException
+   */
   @Override
   public void stop() throws MuleException {
+    LinkedList<Service> servicesWithSpecialStoppingOrder = new LinkedList<>();
```
Comment (marianogonzalez) [inline modules/service/src/main/java/org/mule/runtime/module/service/internal/manager/MuleServiceManager.java]: use `new ArrayList<>(2)` or simply two variables.

File: modules/service/src/main/java/org/mule/runtime/module/service/internal/manager/MuleServiceManager.java [hunk 2]
```diff
@@ -67,15 +70,35 @@ private void startServices() throws MuleException {
...
-        LOGGER.warn(format("Failed to stop service '%s': %s", service.getName(), e.getMessage()), e);
+      if (service.getName().equals(SCHEDULER_SERVICE_ARTIFACT_ID)) {
+        servicesWithSpecialStoppingOrder.addLast(service);
+      } else if (service.getName().contains(HTTP_SERVICE_ARTIFACT_PREFIX)) {
+        servicesWithSpecialStoppingOrder.addFirst(service);
+      } else {
+        doStopService(service);
       }
     }
+    for (Service service : servicesWithSpecialStoppingOrder) {
```
Comment (marianogonzalez) [inline modules/service/src/main/java/org/mule/runtime/module/service/internal/manager/MuleServiceManager.java]: you can leave it as it is if you want.. Keep in mind that you could have used `servicesWithSpecialStoppingOrder.forEach(this::doStopService)` instead

File: modules/service/src/main/java/org/mule/runtime/module/service/internal/manager/MuleServiceManager.java (hunk 1 above)
Comment (plbalbi) [reply modules/service/src/main/java/org/mule/runtime/module/service/internal/manager/MuleServiceManager.java]: Changed to Optionals. Solution seems cleaner.

Comment (plbalbi) [reply modules/service/src/main/java/org/mule/runtime/module/service/internal/manager/MuleServiceManager.java]: Back to null.

**Pull request merged**